ci-operator/step-registry/osd-ccs/fleet-manager/sc/provision/osd-ccs-fleet-manager-sc-provision-commands.sh

```diff
@@ -4,28 +4,89 @@ set -o nounset
+      echo "service cluster state: ${cluster_state}"
+      if [[ "${cluster_state}" == "ready" ]]; then
+        echo "service cluster reported as ready"
+        break
+      fi
+    done
+    return 0
+}
+
+#Set up region
+OSDFM_REGION=${LEASED_RESOURCE}
+echo "region: ${LEASED_RESOURCE}"
+if [[ "${OSDFM_REGION}" != "ap-northeast-1" ]]; then
+  echo "${OSDFM_REGION} is not ap-northeast-1, exit"
+  exit 1
+fi
 
 ## Configure aws
-#CLOUD_PROVIDER_REGION=${LEASED_RESOURCE}
-#AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
-#if [[ -f "${AWSCRED}" ]]; then
-#  export AWS_SHARED_CREDENTIALS_FILE="${AWSCRED}"
-#  export AWS_DEFAULT_REGION="${CLOUD_PROVIDER_REGION}"
-#else
-#  echo "Did not find compatible cloud provider cluster_profile"
-#  exit 1
-#fi
-
-
-# Log in
-OCM_VERSION=$(ocm version)
-OCM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
-echo "Logging into ${OCM_LOGIN_ENV} with offline token using ocm cli ${OCM_VERSION}"
-ocm login --url "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
+if [[ -f "${AWSCRED}" ]]; then
+  export AWS_SHARED_CREDENTIALS_FILE="${AWSCRED}"
+  export AWS_DEFAULT_REGION="${OSDFM_REGION}"
+else
+  echo "Did not find compatible cloud provider cluster_profile"
+  exit 1
+fi
 
-# Log in
+# Log in with OSDFM token
 OCM_VERSION=$(ocm version)
-OCM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/fleetmanager-token")
+OSDFM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/fleetmanager-token")
 echo "Logging into ${OCM_LOGIN_ENV} with offline token using ocm cli ${OCM_VERSION}"
-ocm login --url "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+if [[ ! -z "${OSDFM_TOKEN}" ]]; then
+  echo "Logging into ${OCM_LOGIN_ENV} with osdfm offline token"
+  ocm login --url "${OCM_LOGIN_ENV}" --token "${OSDFM_TOKEN}"
+  if [ $? -ne 0 ]; then
+    echo "Login failed"
+    exit 1
+  fi
+else
+  echo "Cannot login! You need to specify the offline token OSDFM_TOKEN!"
+  exit 1
+fi
+
+# Check if SC already exists
+sc_cluster_num=$(ocm get /api/osd_fleet_mgmt/v1/service_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.total')
+if ((${sc_cluster_num} > 0)); then
+  sc_cluster_id=$(ocm get /api/osd_fleet_mgmt/v1/service_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.items[0].id')
+  echo "Service Cluster '${sc_cluster_id}' already exists in region '${OSDFM_REGION}', exit"
+  exit 1
+fi
+
+# Create SC
+sc_cluster_id=$(echo '{"region": "'${OSDFM_REGION}'", "cloud_provider": "aws"}' | ocm post /api/osd_fleet_mgmt/v1/service_clusters | jq -r '.id')
+# Save Service Cluster info
+echo "Service Cluster:${sc_cluster_id}"
+echo "${sc_cluster_id}" >> "${SHARED_DIR}/osd-fm-sc-id"
+
+# Check if Service Cluster is ready
+echo "Waiting for Service Cluster ready..."
+wait_for_cluster
+
+# Print Management Cluster info, there might be multiple MC if 1st failed
+mc_cluster_num=$(ocm get /api/osd_fleet_mgmt/v1/management_clusters -p search="parent.id='${sc_cluster_id}'" | jq -r '.total')
```

Comment:
You are trying to get a 'ready' MC, why not add the filter in the search `-p search="parent.id='${sc_cluster_id}' and status is 'ready'" -p size=1`?

Besides, you need to add a judgment to treat if there is no 'ready' MC.

Reply:
Good idea! I prints all MC info to detect any failed MC in my design, but right now I can also get to know this info in MC deletion so I changed as your comment, thanks